File: shared/domain/personal_repository.go
package domain

import "github.com/SeijiShii/home-visit-suite/shared/domain/models"

// PersonalRepository は個人データ（DeviceDB格納）の永続化インターフェース。
type PersonalRepository interface {
	// PersonalNote
	GetPersonalNote(visitRecordID string) (*models.PersonalNote, error)
	SavePersonalNote(note *models.PersonalNote) error
	DeletePersonalNote(id string) error

	// PersonalTag
	ListPersonalTags() ([]models.PersonalTag, error)
	SavePersonalTag(tag *models.PersonalTag) error
	DeletePersonalTag(id string) error

	// PersonalTagAssignment
	ListPersonalTagAssignments(visitRecordID string) ([]models.PersonalTagAssignment, error)
	SavePersonalTagAssignment(a *models.PersonalTagAssignment) error
	DeletePersonalTagAssignment(id string) error

	// AppSettings (key-value)

	// GetHiddenTipKeys は非表示化されたヘルプ tip キーの一覧を返す。
	// 初期状態では空配列を返し、エラーにはしない。
	GetHiddenTipKeys() ([]string, error)
	// AddHiddenTipKey は指定キーを非表示リストに追加する。既に存在する場合は何もしない。
	AddHiddenTipKey(key string) error
	// ClearHiddenTipKeys は非表示リストを全消去する。
	ClearHiddenTipKeys() error

	// GetLocale は保存済みの UI 言語コード ("ja"/"en" 等) を返す。未設定の場合は空文字。
	GetLocale() (string, error)
	// SetLocale は UI 言語コードを保存する。
	SetLocale(locale string) error
}

File: pwa/src/domain/repositories/personal-repository.ts
// 個人データ（DeviceDB格納）の永続化インターフェース。
// 参照実装: shared/domain/personal_repository.go

import type {
  PersonalNote,
  PersonalTag,
  PersonalTagAssignment,
} from "../models/personal";

export interface PersonalRepository {
  // PersonalNote
  getPersonalNote(visitRecordId: string): Promise<PersonalNote | null>;
  savePersonalNote(note: PersonalNote): Promise<void>;
  deletePersonalNote(id: string): Promise<void>;

  // PersonalTag
  listPersonalTags(): Promise<PersonalTag[]>;
  savePersonalTag(tag: PersonalTag): Promise<void>;
  deletePersonalTag(id: string): Promise<void>;

  // PersonalTagAssignment
  listPersonalTagAssignments(
    visitRecordId: string,
  ): Promise<PersonalTagAssignment[]>;
  savePersonalTagAssignment(a: PersonalTagAssignment): Promise<void>;
  deletePersonalTagAssignment(id: string): Promise<void>;

  // AppSettings (key-value)

  /** 非表示化されたヘルプ tip キーの一覧を返す。初期状態では空配列を返し、エラーにはしない。 */
  getHiddenTipKeys(): Promise<string[]>;
  /** 指定キーを非表示リストに追加する。既に存在する場合は何もしない。 */
  addHiddenTipKey(key: string): Promise<void>;
  /** 非表示リストを全消去する。 */
  clearHiddenTipKeys(): Promise<void>;

  /** 保存済みの UI 言語コード ("ja"/"en" 等) を返す。未設定の場合は空文字。 */
  getLocale(): Promise<string>;
  /** UI 言語コードを保存する。 */
  setLocale(locale: string): Promise<void>;

  /** 区域詳細編集モードの隣接半径(km)を返す。未設定時は 0 を返す。 */
  getAreaDetailRadiusKm(): Promise<number>;
  /** 隣接半径(km)を保存する。 */
  setAreaDetailRadiusKm(km: number): Promise<void>;
}
